### PR TITLE
[ROCm] Support of hipblaslt

### PR DESCRIPTION
**Pull request by ekuznetsov139**
This PR:

* Enables detection of hipblaslt. Hooks it up in place of cublaslt when available. 

* Enables the _FusedMatMul op for ROCm and routes it to hipblaslt. 

* Fixes certain #includes due to their deprecation in ROCm 5.6.

### CODE REVIEW

**Comment by i-chaochen** [comment]
@akuegel would you review our hipblas_lt PR, please? Thanks in advance!

File: tensorflow/compiler/xla/stream_executor/rocm/rocm_dnn.cc [hunk 1]
```diff
@@ -3868,6 +3871,155 @@ bool MIOpenSupport::DoBatchNormalizationBackwardImpl(
   return true;
 }
 
+template <typename T>
+void launchInplaceBiasActivation(hipStream_t stream, void* c_data, const void* bias_data, int activation_mode, uint64_t m, uint64_t n, int64_t ldc, float param);
```
Comment (akuegel) [inline tensorflow/compiler/xla/stream_executor/rocm/rocm_dnn.cc:3875]: There seem to be quite a lot of long lines in this file. Can you please reformat to lines of length <= 80?

File: tensorflow/compiler/xla/stream_executor/rocm/rocm_helpers.cu.cc [hunk 2]
```diff
@@ -74,5 +75,78 @@ void rocm_MakeBatchPointers(void* stream, char* base, int stride, int n,
       ptrs_out);
 }
```
Comment (akuegel) [inline tensorflow/compiler/xla/stream_executor/rocm/rocm_helpers.cu.cc:77]: There seem to be quite a lot of long lines in this file. Can you please reformat to lines of length <= 80?

**Review by akuegel — CHANGES_REQUESTED** [review]

**Comment by akuegel** [comment]
@ekuznetsov139 Not sure whether you saw my review comments? I got asked by @i-chaochen about this PR, but I think this now needs the changes from your side until I can approve it.

**Comment by i-chaochen** [comment]
> @ekuznetsov139 Not sure whether you saw my review comments? I got asked by @i-chaochen about this PR, but I think this now needs the changes from your side until I can approve it.

Hi @akuegel  Thanks for the feedback. I am going to split this big PR to smaller ones and I will do the changes based on your feedback as well.

We're going to push it to XLA first. Thanks in advance!

**Comment by gbaned** [comment]
Hi @ekuznetsov139 Can you please rebase your branch and resolve conflicts? Thank you!

**Pull request closed without merging**